sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -60,6 +34,44 @@ export function buildRequestUrl(
   );
 }
 
+function appendQueryParams(url: string, options: RequestParameters = {}) {
+  if (!options.queryParameters) {
+    return url;
+  }
+  let parsedUrl = new URL(url);
+  const queryParams = options.queryParameters;
+  for (const key of Object.keys(queryParams)) {
+    const param = queryParams[key] as any;
+    if (param === undefined || param === null) {
+      continue;
+    }
+    if (!param.toString || typeof param.toString !== "function") {
+      throw new Error(`Query parameters must be able to be represented as string, ${key} can't`);
+    }
+    const value = param.toISOString !== undefined ? param.toISOString() : param.toString();
+    parsedUrl.searchParams.append(key, value);
+  }
+
+  if (options.skipUrlEncoding) {
+    parsedUrl = skipQueryParameterEncoding(parsedUrl);
+  }
+  return parsedUrl.toString();
+}
+
+function skipQueryParameterEncoding(url: URL) {
+  if (!url) {
+    return url;
+  }
+  const searchPieces: string[] = [];
+  for (let [name, value] of url.searchParams) {
+    // QUIRK: searchParams.get will NOT encode the values
```

Comment:
nit, slight rephrase here: 
```suggestion
    // QUIRK: searchParams.get retrieves the values decoded
```